src/main/java/org/apache/mesos/offer/ResourceUtils.java

```diff
@@ -211,6 +218,49 @@ public static ExecutorInfo clearResourceIds(ExecutorInfo executorInfo) {
             .build();
   }
 
+  public static boolean areDifferent(
+          ResourceSpecification oldResourceSpecification,
+          ResourceSpecification newResourceSpecification) {
+
+    Protos.Value oldValue = oldResourceSpecification.getValue();
+    Protos.Value newValue = newResourceSpecification.getValue();
+    if (!ValueUtils.equal(oldValue, newValue)) {
+      LOGGER.info("Values are different.");
```

Comment:
Also, these could print the things being compared: the two values, the two roles, the two principals